yt_dlp/extractor/googledrive.py

```diff
@@ -268,10 +268,9 @@ def add_source_format(urlh):
         subtitles_id = None
         ttsurl = get_value('ttsurl')
         if ttsurl:
-            # the video Id for subtitles will be the last value in the ttsurl
-            # query string
-            subtitles_id = ttsurl.encode().decode(
-                'unicode_escape').split('=')[-1]
+            # the subtitles ID is the vid param of the ttsurl query
+            subtitles_id = urllib.parse.parse_qs(
+                urllib.parse.urlparse(ttsurl).query).get('vid', [None])[0]
```

Comment:
use `yt_dlp.utils.parse_qs`
```suggestion
            subtitles_id = parse_qs(ttsurl).get('vid', [None])[-1]
```